File: BindingNavigatorVBWinForms/Form1.vb
Imports BindingNavigatorVBWinForms.Classes

Public Class Form1
    Private Sub Form1_Shown(sender As Object, e As EventArgs) Handles Me.Shown

        BindingSource1.DataSource = BogusOperations.PeopleList(10)
        DataGridView1.DataSource = BindingSource1

    End Sub
End Class

File: WinFormsApp1/Form1.cs
using WinFormsApp1.Classes;

namespace WinFormsApp1;

public partial class Form1 : Form
{
    public Form1()
    {
        InitializeComponent();
        Shown += OnShown;
    }

    private void OnShown(object? sender, EventArgs e)
    {
        CategoryComboBox.DataSource = DataOperations.Categories();
    }

    private void CurrentSelectionButton_Click(object sender, EventArgs e)
    {
        var category = (Category)CategoryComboBox.SelectedItem;
        MessageBox.Show($"{category.Id} {category.Name}");
    }
}
